packages/eas-cli/src/credentials/ios/appstore/ensureAppExists.ts

```diff
@@ -1,24 +1,78 @@
+import { BundleId, CapabilityType, CapabilityTypeOption } from '@expo/apple-utils';
 import ora from 'ora';
 
-import { AuthCtx } from './authenticate';
+import { AuthCtx, ensureAuthenticatedAsync } from './authenticate';
+import { USE_APPLE_UTILS } from './experimental';
 import { runActionAsync, travelingFastlane } from './fastlane';
 
+export interface EnsureAppExistsOptions {
+  enablePushNotifications?: boolean;
+}
+
 export interface AppLookupParams {
   accountName: string;
   projectName: string;
   bundleIdentifier: string;
 }
 
-export interface EnsureAppExistsOptions {
-  enablePushNotifications?: boolean;
+async function ensureBundleIdExistsAsync(
+  appleCtx: Omit<AuthCtx, 'fastlaneSession'>,
+  { accountName, projectName, bundleIdentifier }: AppLookupParams,
+  options: EnsureAppExistsOptions = {}
+) {
+  let spinner = ora(`Registering Bundle ID "${bundleIdentifier}"`).start();
+
+  try {
+    appleCtx = await ensureAuthenticatedAsync(appleCtx);
```

Comment:
The Apple auth state is pretty complex and this `credentials/ios/appstore` logic would need to be copied over to everything that interacts with Apple through our package. Our apple package also uses its awareness of the auth state to validate when forbidden network requests are made.

Reply:
I'm not sure we are talking about the same thing, I'm suggesting that you move this call here
https://github.com/expo/eas-cli/blob/main/packages/eas-cli/src/credentials/ios/appstore/AppStoreApi.ts#L56

```
  public async ensureAppExistsAsync(
    app: AppLookupParams,
    options?: EnsureAppExistsOptions
  ): Promise<void> {
    let ctx 
    if (USE_APPLE_UTILS) {
      ctx = await ensureAuthenticatedAsync(this.options);
    } else {
      ctx = awiat this.ensureAuthenticatedAsync();
   }
    
    return await ensureAppExistsAsync(ctx, app, options);
  }
```

Or even better to move logic from `ensureAuthenticatedAsync` (your function) into `AppStoreApi.ensureAuthenticatedAsync` method


A lot of issues here comes from the fact that we temporarily support both implementations, so if you disagree about that lets just merge it in current form and we can refactor it after fastlane is removed

Reply:
got it, I see the method `ensureAppExistsAsync` is not being used directly like it is in expo-cli. I removed the extra auth check since I now see that `ensureAuthenticatedAsync` is run in a wrapper method. It would seem that this wrapper causes Fastlane to have an extra auth check, but I suppose that won't be an issue for much longer.